src/shared/actions/archives.js

```diff
@@ -18,25 +18,27 @@ export const removeArchive = createAction(ARCHIVES_REMOVE);
 export const clearArchives = createAction(ARCHIVES_CLEAR);
 
 export const loadArchive = payload => async (dispatch, getState) => {
-  const archive = await showPasswordDialog(
-    password => loadWorkspace(payload, password).catch(err => {
-      const unknownMessage = 'An unknown error has occurred';
-      return Promise.reject(
-        isError(err)
-          ? err.message || unknownMessage
-          : unknownMessage
-      );
-    })
-  );
+  try {
+    const archive = await showPasswordDialog(
+      password => loadWorkspace(payload, password).catch(err => {
+        const unknownMessage = 'An unknown error has occurred';
+        return Promise.reject(
```

Comment:
Why reject? You could just throw another error as you're within a Promise anyhow - it'd work the same way and be a little neater (seeing as you're using await). What about:

```javascript
if (isError(err)) {
  throw err;
} else {
  throw new Error(unknownMessage);
}
```